src/App.js

```diff
@@ -27,13 +45,20 @@ export default class App extends React.Component {
               <ViewSpace />
             </Route>
             <Route path="/survey">
-              <SurveyView />
+              <Survey />
             </Route>
             <Route path="/search">
               <Search />
             </Route>
+            <Route path='/login'>
+              {this.state.loggedIn ? <Redirect to='/'/> : <LandingPage login={this.login}/>}
+            </Route>
             <Route path="/">
-              {/* TODO home */}
+              {this.state.loggedIn ? <HomePage /> : <Redirect to='/login'/>}
+            </Route>
```

Comment:
Temporary Routing. Has some slight bugs that I will probably need help thinking of solutions for.

Reply:
could you elaborate on bugs